pandas/tests/series/indexing/test_indexing.py

```diff
@@ -330,6 +332,14 @@ def test_loc_setitem_all_false_indexer():
     tm.assert_series_equal(ser, expected)
 
 
+def test_reindexing_values():
+    # GH 47055
+    s = Series([1.0, NA], dtype=Float64Dtype())
+    s_reindex = s.reindex(range(3))
```

Comment:
1. Could you move this test to `pandas/tests/series/methods/test_reindex.py`?
2. Could you mention in the test name that it involves Float64 with NA?
3. Could you also test that a warning isnt raised like

```
tm.assert_produces_warning(None):
    result = np.log(s_reindex)
    expected = ...
```